cloudinit/distros/networking.py

```diff
@@ -0,0 +1,137 @@
+import abc
+
+from cloudinit import net
+
+
+# Type aliases (https://docs.python.org/3/library/typing.html#type-aliases),
+# used to make the signatures of methods a little clearer
+DeviceName = str
+NetworkConfig = dict
+
+
+class Networking(metaclass=abc.ABCMeta):
+    """The root of tne Networking hierarchy in cloud-init.
+
+    This is part of an ongoing refactor in the cloud-init codebase, for more
+    details see "``cloudinit.net`` -> ``cloudinit.distros.networking``
+    Hierarchy" in HACKING.rst for full details.
+    """
+
+    def _get_current_rename_info(self) -> dict:
+        return net._get_current_rename_info()
+
+    def _rename_interfaces(self, renames: list, *, current_info=None) -> None:
+        return net._rename_interfaces(renames, current_info=current_info)
+
+    def apply_network_config_names(self, netcfg: NetworkConfig) -> None:
+        return net.apply_network_config_names(netcfg)
```

Comment:
Can we also raise an `NotImplementedError` by default?

Reply:
When we convert this to an abstractmethod (as per https://github.com/canonical/cloud-init/pull/391/files#diff-0abd02cf07a2406439d0d263b886ebcbR434), then we won't be able to instantiate `Networking` subclasses without an implementation. If the appropriate default behaviour on BSD is to `raise NotImplementedError`, then that should be the implementation in `BSDNetworking`.

(Of course, any callers will also need to cope with that exception being raised, so it may be the case that a different default behaviour makes sense; it may be acceptable to make it a noop, for example.)

Reply:
i think if the exception wasn't BSD specific, there might be more of an incentive to actually catch or handle it

Reply:
Well, this method will have an implementation on `LinuxNetworking`, so regardless of whether the exception is raised in `Networking.apply_network_config_names` or in `BSDNetworking.apply_network_config_names`, it will only be raised on BSD systems.

Reply:
+1 this works for me thanks for addressing the concern @OddBloke